### ISSUE TITLE
Convert Long to Duration

### ISSUE THREAD

**Issue by maxthoursie** [juxt/tick#46]
I miss being able to do `(t/hours 2)` and I find `(t/new-duration i :hours)` a little to verbose.

I can easily get what I want by extending java.lang.Long like:
```clj
(extend-protocol tick.core/ITimeLength
  java.lang.Long
  (hours [i] (t/new-duration i :hours)))

(t/hours 2) => #time/duration"PT2H"
```

Is there any downsides doing that? Could we include it in tick?

Thank you _so_ much for creating tick and the underlying cljc.java-time!

**Comment by henryw374** [comment]
Hi, sorry I saw this then it dropped off my radar. 

The thing I think is a bit strange about this is that ITimeLength would be used for construction and extraction:

`(t/hours (t/hours 1)) ` => 1

Perhaps a set of fns like `new-duration-hours`, `new-duration-millis`. Needing better/shorter names perhaps ;-). 

@malcolmsparks thoughts?

FYI I started impl using the protocol https://github.com/juxt/tick/commits/construction-using-itimelength

**Comment by malcolmsparks** [comment]
I agree with @henryw374 here. The problem is that of overloading semantics. A function should only mean one thing, even if it has multiple arities. In the case of `t/hours`, it is taken to be a coercion from an existing time quantity rather than a constructor.

Perhaps `(of-hours 1)` ?
